packages/twenty-server/src/core/auth/services/auth.service.ts

```diff
@@ -157,23 +161,57 @@ export class AuthService {
       imagePath = paths[0];
     }
 
-    const userToCreate = this.userRepository.create({
-      email: email,
-      firstName: firstName,
-      lastName: lastName,
-      defaultAvatarUrl: imagePath,
-      canImpersonate: false,
-      passwordHash,
-      defaultWorkspace: workspace,
-    });
+    if (existingUser && workspaceInviteHash) {
+      const userWorkspaceExisits =
+        await this.userWorkspaceService.checkUserWorkspaceExists(
+          existingUser.id,
+          workspace.id,
+        );
 
-    const user = await this.userRepository.save(userToCreate);
+      if (userWorkspaceExisits) return existingUser;
 
-    if (workspaceInviteHash) {
-      await this.userService.createWorkspaceMember(user);
-    }
+      const userWorkspace = await this.userWorkspaceService.create(
+        existingUser.id,
+        workspace.id,
+      );
 
-    return user;
+      await this.userWorkspaceService.createWorkspaceMember(
+        workspace.id,
+        existingUser,
+      );
```

Comment:
I think about stopping creating the workspace member at signUp, but only at the createProfile step. What do you think?

Reply:
I think we need it here, this call is executed in the block of `existingUser`, meaning that this user has already has signed-up (that means he has created a profile) and already has a workspace assigned. If this user is joining another workspace, then we need to map it as a member of that workspace, hence I added it here.

Or I might be overthinking this, let me know your thoughts